mono/mini/debugger-agent.c

```diff
@@ -4847,6 +4856,35 @@ ss_clear_for_assembly (SingleStepReq *req, MonoAssembly *assembly)
 	}
 }
 
+/*
+ * This takes a lot of locks and stuff. Do this at the end, after
+ * other things have dumped us, so that getting stuck here won't
+ * prevent seeing other crash information
+ */
+void
+mono_debugger_agent_send_crash (char *jsonDump, MonoStackHash *hashes, int pause)
+{
+	int suspend_policy;
+	GSList *events;
+	EventInfo ei;
+
+	if (!agent_config.enabled)
+		return;
+
+	// It doesn't make sense to wait for lldb/gdb to finish if we're not
+	// actually enabled. Therefore we do the wait here.
+	sleep (pause);
+
+	mono_loader_lock ();
```

Comment:
This needs to check that protocol version.

Reply:
I was about to ask where the downgrade happened, thanks

Reply:
The check was added